src/app/app.module.ts

```diff
@@ -8,10 +8,23 @@ import { App } from './app.component';
 import { JournalProvider } from '../providers/journal/journal';
 import { HttpClientModule } from '@angular/common/http';
 import { AppConfigProvider } from '../providers/app-config/app-config';
+import { StoreModule } from '@ngrx/store';
+import { journalReducer } from '../store/journal.reducer';
+import { StoreDevtoolsModule } from '@ngrx/store-devtools';
+import { EffectsModule } from '@ngrx/effects';
+import { JournalEffects } from '../effects/journal.effects';
 
 @NgModule({
   declarations: [App],
-  imports: [BrowserModule, HttpClientModule, IonicModule.forRoot(App , {mode: 'ios'})],
+  imports: [
+    BrowserModule,
+    HttpClientModule,
+    IonicModule.forRoot(App, { mode: 'ios' }),
+    StoreModule.forRoot({
+      journal: journalReducer
+    }),
```

Comment:
Should we be putting the actions, reducers and effects within the page modules and using feature state instead of a global state?

Link for examples of both - https://offering.solutions/blog/articles/2018/02/10/separating-state-into-angular-modules-with-ngrx/

Reply:
Made the changes. Now we have imported the reducer and effect in the page module.